tests/test_karma.py

```diff
@@ -93,3 +93,34 @@ def test_change_karma_bot_self():
         karma.change_karma(-3)
         == "Not cool pybob lowering my karma to 12, but you are probably right, I will work harder next time"
     )
+
+
+@pytest.mark.parametrize(
+    "karma_giver, channel_id, karma_changes, expected",
+    [
+        (
+            "ABC123",
+            "FOO123",
+            [("<@EFG123>", "++"), ("<@XYZ123>", "++")],
+            [
+                "Julian Sequeira's karma increased to 125",
+                "clamytoe's karma increased to 422",
+            ],
+        ),
+        (
+            "XYZ123",
+            "FOO123",
+            [("<@ABC123>", "+++"), ("<@EFG123>", "++++")],
+            [
+                "pybob's karma increased to 395",
+                "Julian Sequeira's karma increased to 127",
+            ],
+        ),
+    ],
+)
+@pytest.mark.usefixtures("save_transaction_disabled", "mock_filled_db_session")
+def test_process_karma_changes(karma_giver, channel_id, karma_changes, expected):
+    messages = process_karma_changes(karma_giver, channel_id, karma_changes)
+
+    for actual, expected in zip(messages, expected):
```

Comment:
do we need this loop? can't we just do `assertt messages == expected`?

Reply:
would be an option. i thought it would be easier as the failure in pytest would show the respective line and not point to something inside a potentially large list (if we extend cases to x5 karma in one)

Reply:
thanks